// polotno/side-panel/side-panel.js
import e from"react";import{observer as t}from"mobx-react-lite";import{Icon as n}from"@blueprintjs/core";import{Video as o,Control as a,NewTextBox as l,Media as i,CloudUpload as r,LayoutGrid as s,Layers as m,Fullscreen as c,More as p}from"@blueprintjs/icons";import d from"../utils/styled.js";import{useMobile as b,mobileStyle as f}from"../utils/screen.js";import{flags as E}from"../utils/flags.js";import u from"@meronex/icons/fa/FaShapes.js";import h from"@meronex/icons/fd/FdPageMultiple.js";import{t as g}from"../utils/l10n.js";import{SectionTab as x}from"./tab-button.js";export{SectionTab}from"./tab-button.js";export{ImagesGrid}from"./images-grid.js";import{TextPanel as S}from"./text-panel.js";import{SizePanel as P}from"./size-panel.js";import{UploadPanel as v}from"./upload-panel.js";import{PhotosPanel as j}from"./photos-panel.js";import{BackgroundPanel as w}from"./background-panel.js";import{ElementsPanel as T}from"./elements-panel.js";import{PagesPanel as k}from"./pages-panel.js";import{TemplatesPanel as C}from"./templates-panel.js";import{LayersPanel as N}from"./layers-panel.js";import{VideosPanel as O}from"./videos-panel.js";import{ImageClipPanel as y}from"./image-clip-panel.js";import{AnimationsPanel as L}from"./animations-panel.js";import{EffectsPanel as z}from"./effects-panel.js";export const TemplatesSection={name:"templates",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.templates")},t),e.createElement(n,{icon:e.createElement(a,null)}))),Panel:({store:t})=>e.createElement(C,{store:t})};export const TextSection={name:"text",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.text")},t),e.createElement(n,{icon:e.createElement(l,null)}))),Panel:({store:t})=>e.createElement(S,{store:t})};export const PhotosSection={name:"photos",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.photos")},t),e.createElement(n,{icon:e.createElement(i,null)}))),Panel:({store:t})=>e.createElement(j,{store:t})};export const ElementsSection={name:"elements",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.elements"),iconSize:16},t),e.createElement("span",{className:"bp5-icon"},e.createElement(u,null)))),Panel:({store:t})=>e.createElement(T,{store:t})};export const UploadSection={name:"upload",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.upload")},t),e.createElement(n,{icon:e.createElement(r,null)}))),Panel:({store:t})=>e.createElement(v,{store:t})};export const BackgroundSection={name:"background",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.background")},t),e.createElement(n,{icon:e.createElement(s,null)}))),Panel:({store:t})=>e.createElement(w,{store:t})};export const PagesSection={name:"pages",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.pages")},t),e.createElement(h,null))),Panel:({store:t})=>e.createElement(k,{store:t}),visibleInList:!1};export const LayersSection={name:"layers",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.layers")},t),e.createElement(n,{icon:e.createElement(m,null)}))),Panel:({store:t})=>e.createElement(N,{store:t})};export const SizeSection={name:"size",Tab:t(t=>e.createElement(x,Object.assign({name:g("sidePanel.resize")},t),e.createElement(n,{icon:e.createElement(c,null)}))),Panel:({store:t})=>e.createElement(P,{store:t})};export const VideosSection={name:"videos",Tab:t=>e.createElement(x,Object.assign({name:g("sidePanel.videos")},t),e.createElement(o,null)),Panel:O};t(t=>e.createElement(x,Object.assign({name:g("sidePanel.more")},t),e.createElement(n,{icon:e.createElement(p,null)})));export const DEFAULT_SECTIONS=[TemplatesSection,TextSection,PhotosSection,ElementsSection,UploadSection,BackgroundSection,LayersSection,SizeSection];const I={name:"image-clip",Tab:()=>null,Panel:y},R={name:"effects",Tab:()=>null,Panel:z},$={name:"animation",Tab:()=>null,Panel:L};export const INTERNAL_SECTIONS=[I,R];const A=d("div")`
  display: flex;
  height: 100% !important;
  padding: 0px !important;
  position: relative;

  &.bp5-navbar {
    box-shadow: none;
  }

  ${f("\n    height: auto !important;\n    width: 100%;\n    position: relative;\n  ")}
`,F=d("div",e.forwardRef)`
  @media screen and (min-width: 501px) {
    overflow-y: auto;
    overflow-x: hidden;
    min-width: 72px;
  }
  ${f("\n    width: 100%;\n    overflow: auto;\n  ")}
`,M=d("div",e.forwardRef)`
  display: flex;
  flex-direction: column;

  ${f("\n    flex-direction: row;\n    min-width: min-content;\n  ")}
`,U=d("div")`
  padding: 10px 10px 0px 10px !important;
  height: 100% !important;

  &.bp5-navbar {
    width: 350px;
  }

  &.bp5-navbar.collapsed {
    width: 0px;
  }

  ${f("\n    &.bp5-navbar {\n      position: absolute;\n      bottom: 54px;\n      z-index: 100;\n      height: 50vh !important;\n      width: 100%;\n    }\n  ")}
`,_=d("div")`
  display: none;

  ${f("\n    position: absolute;\n    bottom: 72px;\n    display: block;\n    width: 100vw;\n    height: 100vh;\n    background-color: rgba(0, 0, 0, 0.05);\n  ")}
`,B=d("div")`
  position: absolute;
  right: -14px;
  top: 50%;
  height: 96px;
  width: 15px;
  fill: white;
  cursor: pointer;
  z-index: 10;

  .bp5-dark & {
    right: -13px;
  }

  & .stroke {
    stroke: rgba(0, 0, 0, 0.3);
    fill: none;
  }

  & .fill {
    fill: white;
  }

  .bp5-dark & .fill {
    fill: #2f343c;
  }

  & .pointer {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%) scaleY(2);
    font-size: 0.5rem;
    color: rgba(171, 179, 191, 0.6);
  }

  .bp5-dark & .pointer {
    color: rgba(171, 179, 191, 0.6);
  }

  &:hover .pointer {
    color: black;
  }

  .bp5-dark &:hover .pointer {
    color: white;
  }

  ${f("\n    display: none;\n  ")}
`,V=({onClick:t})=>e.createElement(B,{onClick:t},e.createElement("svg",{width:"15",height:"96",viewBox:"0 0 16 96",xmlns:"http://www.w3.org/2000/svg"},e.createElement("path",{className:"fill stroke",d:"M 1 7 C 2 9 4 10 7.2 13.28 C 7.45 13.4625 7.6 13.6 7.7 13.8048 L 7.8 13.8 C 9.8 15.8 11.6 17.6 12.9 19.7 C 14 21.6 14.7 23.9 14.9 27 H 15 V 68 C 15 71.7 14.3 74.3 13 76.6 C 11.7 78.8 9.9 80.5 7.8 82.6344 L 7.79 82.6 C 7.6 82.8 7.4507 83 7.2729 83.2127 C 3.9102 86.5228 2 87 1 90"})),e.createElement("div",{className:"pointer"},"<"));export const SidePanel=t(({store:t,sections:n,defaultSection:o="photos"})=>{var a;(()=>{const n=e.useRef(!0);n.current&&(n.current=!1,t.openSidePanel(o))})();const l=b();e.useLayoutEffect(()=>{l?t.openSidePanel(""):t.openSidePanel(o)},[l]);const i=[...n||DEFAULT_SECTIONS];if(INTERNAL_SECTIONS.forEach(e=>{i.find(t=>t.name===e.name)||i.push(e)}),E.animationsEnabled){if(!i.find(e=>"videos"===e.name)){const e=Math.floor(i.length/2);i.splice(e,0,VideosSection)}if(!i.find(e=>"animations"===e.name)){const e=Math.floor(i.length/2);i.splice(e,0,$)}}const r=i.filter(e=>{var t;return null===(t=e.visibleInList)||void 0===t||t}),s=null===(a=i.find(e=>e.name===t.openedSidePanel))||void 0===a?void 0:a.Panel,m=e.useRef(null);return e.createElement(A,{className:"bp5-navbar polotno-side-panel"+(t.openedSidePanel?"":" collapsed")},e.createElement(F,{ref:m,className:"polotno-side-tabs-container"},e.createElement(M,{className:"polotno-side-tabs-inner"},r.map(({name:n,Tab:o})=>e.createElement(o,{key:n,active:n===t.openedSidePanel,onClick:()=>{n===t.openedSidePanel&&l?t.openSidePanel(""):t.openSidePanel(n)}})))),s&&e.createElement(U,{className:"bp5-navbar polotno-panel-container",onClick:e=>{const n=e.target.closest(".polotno-close-panel"),o=e.target.closest(".polotno-mobile");n&&(l||o)&&t.openSidePanel("")}},e.createElement(s,{store:t})),t.openedSidePanel&&e.createElement(_,{onClick:()=>t.openSidePanel("")}),t.openedSidePanel&&e.createElement(V,{onClick:()=>t.openSidePanel("")}))});export default SidePanel;
